x/distribution/keeper/abci_test.go

```diff
@@ -0,0 +1,644 @@
+
+	votes := []abci.VoteInfo{
+		{
+			Validator: abci.Validator{
+				Address: valConsPk0.Address(),
+				Power:   100,
+			},
+		},
+		{
+			Validator: abci.Validator{
+				Address: valConsPk1.Address(),
+				Power:   100,
+			},
+		},
+	}
+	ctx = ctx.WithVoteInfos(votes).WithBlockHeight(2)
+
+	feePoolBefore, err := ts.distrKeeper.FeePool.Get(ctx)
+	require.NoError(t, err)
+	err = ts.distrKeeper.BeginBlocker(ctx)
+	require.NoError(t, err)
+	feePoolAfter, err := ts.distrKeeper.FeePool.Get(ctx)
+	require.NoError(t, err)
+
+	require.False(t, feePoolBefore.CommunityPool.Equal(feePoolAfter.CommunityPool), fmt.Sprintf("before: %s, after: %s", feePoolBefore.CommunityPool.String(), feePoolAfter.CommunityPool.String()))
+
+	// 98 outstanding rewards (100 less 2 to community pool)
+	val0OutstandingRewards, err = ts.distrKeeper.GetValidatorOutstandingRewards(ctx, valAddr0)
+	require.NoError(t, err)
+	require.Equal(t, sdk.DecCoins{{Denom: sdk.DefaultBondDenom, Amount: math.LegacyNewDecWithPrec(490, 1)}}, val0OutstandingRewards.Rewards)
+
+	val1OutstandingRewards, err = ts.distrKeeper.GetValidatorOutstandingRewards(ctx, valAddr1)
+	require.NoError(t, err)
+	require.Equal(t, sdk.DecCoins{{Denom: sdk.DefaultBondDenom, Amount: math.LegacyNewDecWithPrec(490, 1)}}, val1OutstandingRewards.Rewards)
+
+	// 2 community pool coins
+	feePool, err = ts.distrKeeper.FeePool.Get(ctx)
+	require.NoError(t, err)
+	require.Equal(t, sdk.DecCoins{{Denom: sdk.DefaultBondDenom, Amount: math.LegacyNewDec(2)}}, feePool.CommunityPool)
+
+	// 50% commission for first proposer, (0.5 * 98%) * 100 / 2 = 23.25
+	val0Commission, err = ts.distrKeeper.GetValidatorAccumulatedCommission(ctx, valAddr0)
+	require.NoError(t, err)
+	require.Equal(t, sdk.DecCoins{{Denom: sdk.DefaultBondDenom, Amount: math.LegacyNewDecWithPrec(2450, 2)}}, val0Commission.Commission)
+
+	// zero commission for second proposer
+	val1Commission, err = ts.distrKeeper.GetValidatorAccumulatedCommission(ctx, valAddr1)
+	require.NoError(t, err)
+	require.True(t, val1Commission.Commission.IsZero())
+
+	// just staking.proportional for first proposer less commission = (0.5 * 98%) * 100 / 2 = 24.50
+	val0CurrentRewards, err = ts.distrKeeper.GetValidatorCurrentRewards(ctx, valAddr0)
+	require.NoError(t, err)
+	require.Equal(t, sdk.DecCoins{{Denom: sdk.DefaultBondDenom, Amount: math.LegacyNewDecWithPrec(2450, 2)}}, val0CurrentRewards.Rewards)
+
+	// proposer reward + staking.proportional for second proposer = (0.5 * (98%)) * 100 = 49
+	val1CurrentRewards, err = ts.distrKeeper.GetValidatorCurrentRewards(ctx, valAddr1)
+	require.NoError(t, err)
+	require.Equal(t, sdk.DecCoins{{Denom: sdk.DefaultBondDenom, Amount: math.LegacyNewDecWithPrec(490, 1)}}, val1CurrentRewards.Rewards)
+
+	// check cons address
+	got, err := ts.distrKeeper.GetPreviousProposerConsAddr(ctx)
+	require.NoError(t, err)
+	require.Equal(t, testProposerAddress, got)
+}
+
+func TestBeginBlockTruncation(t *testing.T) {
```

Comment:
same here. what is truncation? im not sure exactly what scenario is being tested here vs the other

Reply:
updated test names and added some more comments and scenario descriptions.

These tests were copies of the exist tests found [here](https://github.com/cosmos/cosmos-sdk/blob/0fbbf26f63db0bb0b2943f1a8f25950bf8bc802f/x/distribution/keeper/allocation_test.go#L85) but encompassed the beginblocker